### ISSUE TITLE
The label value changed

### ISSUE THREAD

**Issue by wendy-xiaozong** [fepegar/torchio#194]
My label value is probability between [0, 1], however, the output of my validation label value becomes this:
max targets: 0.18760673701763153
min targets: -77.21928405761719
my training set and validation set both changed, but training set change smaller, it become like this:
training max targets: 2.3406858444213867
training min targets: -0.5744394063949585
this is my transform code:
```python
def get_train_transforms() -> Compose:
    training_transform = Compose([
        ToSqueeze(),
        ToCanonical(),
        ZNormalization(masking_method=ZNormalization.mean),  # Subtract mean and divide by standard deviation.
        RescaleIntensity((0, 1)),  # so that there are no negative values for RandomMotion
        RandomMotion(),
        RandomBiasField(),
        RandomNoise(),
        RandomFlip(axes=(0,)),
        OneOf({
            RandomAffine(): 0.8,
            RandomElasticDeformation(): 0.2,
        }),
        ToResize()
    ])

    return training_transform


def get_val_transform() -> Compose:
    validation_transform = Compose([
        ToSqueeze(),
        ZNormalization(masking_method=ZNormalization.mean),
        ToCanonical(),
    ])
    return validation_transform
```
`ToResize()`, and `ToSqueeze()` is two custom classes:
```python
class ToSqueeze(Transform):
    def apply_transform(self, sample: Subject) -> dict:
        for image_dict in sample.get_images(intensity_only=False):
            image_dict[DATA] = image_dict[DATA].squeeze().unsqueeze(0)
        return sample


class ToResize(Transform):
    def apply_transform(self, sample: Subject) -> dict:
        for image_dict in sample.get_images(intensity_only=False):
            image_dict[DATA] = F.interpolate(image_dict[DATA].unsqueeze(0), size=(SIZE, SIZE, SIZE))
            image_dict[DATA] = image_dict[DATA].squeeze(0)
        return sample
```
I have no idea why this happens. Thanks for helping!

**Comment by fepegar** [comment]
How do you initialize your `Image`s? Are you setting the type of your labels to `torchio.LABEL`?

**Comment by wendy-xiaozong** [comment]
What do you mean by initializing `Image`s?
Yes, I have set the labels to `torchio.LABEL`

```python
    subjects = [
        tio.Subject(
                img=tio.Image(path=mri.img_path, label=tio.INTENSITY),
                label=tio.Image(path=mri.label_path, label=tio.LABEL),
            ) for mri in get_path(datasets)
    ]
```
And I get the image like this:
```python
inputs, targets = batch["img"][DATA], batch["label"][DATA]
```

**Comment by fepegar** [comment]
I haven't been able to reproduce the problem with this code:

```python
SIZE = 100

# [Definitions of your functions and classes here]

subject = torchio.datasets.Colin27()
train_dataset = torchio.ImagesDataset([subject], transform=get_train_transforms())
val_dataset = torchio.ImagesDataset([subject], transform=get_val_transform())

train_sample = train_dataset[0]
val_sample = val_dataset[0]

print('Training range:', train_sample.brain.data.min().item(), train_sample.brain.data.max().item())
print('Validation range:', val_sample.brain.data.min().item(), val_sample.brain.data.max().item())
```

Output:

```
Training range: 0.0 1.0
Validation range: 0.0 1.0 
```


Do you have a minimal working example + some data that you can share? I can give you my email if you prefer.

**Comment by fepegar** [comment]
> What do you mean by initializing Images?

I guess I should've said _instantiate_ instead of _initialize_.

**Comment by fepegar** [comment]
By the way, do you think a `Resize` transform in TorchIO would be useful? See discussion in https://github.com/fepegar/torchio/issues/193.

**Comment by wendy-xiaozong** [comment]
> Do you have a minimal working example + some data that you can share? I can give you my email if you prefer.

Can you give me your email, I do write a minimal working example with several data.

Maybe because I am using the pytorch-lightning, I do not receive this warning before. maybe it is relative?
```
/home/jq/.local/lib/python3.6/site-packages/numpy/lib/function_base.py:3942: RuntimeWarning: invalid value encountered in multiply
  x2 = take(ap, indices_above, axis=axis) * weights_above
/home/jq/.local/lib/python3.6/site-packages/torchio/transforms/augmentation/spatial/random_affine.py:265: RuntimeWarning: invalid value encountered in less
  values = borders_flat[borders_flat < threshold]
```

**Comment by fepegar** [comment]
> Can you give me your email, I do write a minimal working example with several data.

Great! It's fernando.perezgarcia.17@ucl.ac.uk.

> Maybe because I am using the pytorch-lightning, I do not receive this warning before. maybe it is relative?

I also just started using it. I hope it's not ignoring warnings.

However, I think I know where the problem might be. The default pad value for `RandomAffine` is computed using values at the border. I think that still happens for `LABEL` images, but it shouldn't. Can you confirm that your labels have positive values on the image border? I'll try on my side as well.

**Comment by fepegar** [comment]
Well, I just tried and that doesn't seem to be the problem. Please send me that code and data and I'll debug locally.

**Comment by wendy-xiaozong** [comment]
> By the way, do you think a `Resize` transform in TorchIO would be useful? See discussion in #193.

I have read this issue, I do think maybe a `Resize` would help me because I am using Nifti file format in these dataset: [CC359 dataset](https://sites.google.com/view/calgary-campinas-dataset/home), [ADNI dataset](https://doi.gin.g-node.org/10.12751/g-node.aa605a/). And in this dataset, the Nifti file have different image sizes. However, I need the image size to be all the same so I could put them into a batch. Dunno `Resample` still could help in this situation.

**Comment by fepegar** [comment]
If you just resize the images, your voxel spacing (resolution) will change, you probably don't want that. You might want to switch to using `Resample` + `CropOrPad` instead.

**Comment by wendy-xiaozong** [comment]
> If you just resize the images, your voxel spacing (resolution) will change, you probably don't want that. You might want to switch to using `Resample` + `CropOrPad` instead.

That's a good idea. I definitely would do that way. Thanks for it! I am still waiting for the file to upload to onedrive, wait a minute

**Comment by fepegar** [comment]
This is the problem:

```python
img=tio.Image(path=mri.img_path, label=tio.INTENSITY)
label=tio.Image(path=mri.label_path, label=tio.LABEL)
```

The right keyword is not `label`, but `type`. The problem was that your label images were being instantiated as intensity images, and `ZNormalization` was being applied to them. As you selected `mean` as masking method, the values used to compute the mean and std were all ones. Therefore, infinite values were generated at the output of znorm and this was propagated until a library that is less happy about values such as nan or inf raised an exception.

Asking NumPy to be pickier would have helped finding the issue: https://stackoverflow.com/a/22671908/3956024. I might enable that flag when applying transforms, to improve robustness.

P.S.: always [gunzip](https://www.geeksforgeeks.org/gunzip-command-in-linux-with-examples/) your .nii images before sending them :)

**Comment by fepegar** [comment]
This is the problem:

```python
img=tio.Image(path=mri.img_path, label=tio.INTENSITY)
label=tio.Image(path=mri.label_path, label=tio.LABEL)
```

The right keyword is not `label`, but `type`. The problem was that your label images were being instantiated as intensity images, and `ZNormalization` was being applied to them. As you selected `mean` as masking method, the values used to compute the mean and std were all ones. Therefore, infinite values were generated at the output of znorm and this was propagated until a library that is less happy about values such as nan or inf raised an exception.

Asking NumPy to be pickier would have helped finding the issue: https://stackoverflow.com/a/22671908/3956024. I might enable that flag when applying transforms, to improve robustness.

P.S.: always [gunzip](https://www.geeksforgeeks.org/gunzip-command-in-linux-with-examples/) your .nii images before sending them :)

**Comment by wendy-xiaozong** [comment]
> This is the problem:
> 
> ```python
> img=tio.Image(path=mri.img_path, label=tio.INTENSITY)
> label=tio.Image(path=mri.label_path, label=tio.LABEL)
> ```
> 
> The right keyword is not `label`, but `type`. The problem was that your label images were being instantiated as intensity images, and `ZNormalization` was being applied to them. As you selected `mean` as masking method, the values used to compute the mean and std were all ones, therefore the std is 0 and dividing by it yields infinity. Therefore, infinite values were generated at the output of znorm and this was propagated until a library that is less happy about values such as nan or inf raised an exception.
> 
> Asking NumPy to be pickier would have helped finding the issue: https://stackoverflow.com/a/22671908/3956024. I might enable that flag when applying transforms, to improve robustness.
> 
> P.S.: always [gunzip](https://www.geeksforgeeks.org/gunzip-command-in-linux-with-examples/) your .nii images before sending them :)

Oh! yes you are right. Thanks for your time.
